### PR TITLE
Punctuation & typo fixes to CHANGELOG

### PR DESCRIPTION
**Pull request by piskvorky**
Also includes commits from `master` (not sure why these are missing from develop -- only one commit is actually new, a8b1585).

### CODE REVIEW

**Comment by piskvorky** [comment]
@menshikh-iv @tmylk I'd also suggest type-settings the release names (version) as a heading.

It should be easier to navigate the (huge) CHANGELOG file if the version are visually delimited like that, as headings.

**Comment by menshikh-iv** [comment]
@piskvorky In the future, I plan to work with our readme/changelog/вdoc. I would like to do more structured and useful readme for new users.

**Comment by piskvorky** [comment]
@menshikh-iv that's fine.

This PR is only about the CHANGELOG -- please format the headings & keep its structure and formatting consistent.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ Unreleased:
 
 ===========
```
Comment (piskvorky) [inline CHANGELOG.md]: This heading doesn't render correctly.

**Pull request merged**